src/compass/utils/validate_cslc.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import json
+import os
+
+import numpy as np
+from osgeo import gdal
+
+
+def cmd_line_parser():
+    """
+    Command line parser
+    """
+
+    parser = argparse.ArgumentParser(description="""
+                                     Validate reference and generated (secondary) S1 CSLC products""",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-r', '--ref-product', type=str, dest='ref_product',
+                        help='Reference CSLC product (i.e., golden dataset)')
+    parser.add_argument('-mr', '--ref-metadata', type=str, dest='ref_metadata',
+                        help='Reference CSLC metadata file (i.e., golden metadata)')
+    parser.add_argument('-s', '--sec-product', type=str, dest='sec_product',
+                        help='Secondary CSLC product to compare with reference')
+    parser.add_argument('-ms', '--sec-metadata', type=str, dest='sec_metadata',
+                        help='Secondary CSLC metadata to compare with reference metadata')
+    return parser.parse_args()
+
+
+def compare_cslc_products(file_ref, file_sec):
+    '''
+    Compare a reference and a newly generated
+    (i.e., secondary) CSLC product
+
+    Parameters
+    ----------
+    file_ref: str
+        File path to reference CSLC product (golden dataset)
+    file_sec: str
+        File path to generated CSLC product to use for comparison
+    '''
+
+    # Check if file paths exits
+    if not os.path.exists(file_ref):
+        print(f'ERROR reference CSLC product not found: {file_ref}')
+        return
+
+    if not os.path.exists(file_sec):
+        print(f'ERROR secondary CSLC product not found: {file_sec}')
+        return
+
+    # Extract some info from reference/secondary CSLC products
+    dataset_ref = gdal.Open(file_ref, gdal.GA_ReadOnly)
+    geotransform_ref = dataset_ref.GetGeoTransform()
+    nbands_ref = dataset_ref.RasterCount
+
+    dataset_sec = gdal.Open(file_sec, gdal.GA_ReadOnly)
+    geotransform_sec = dataset_sec.GetGeoTransform()
+    nbands_sec = dataset_sec.RasterCount
```

Comment:
Could be a nit-pick, but I would suggest taking care of the loaded gdal rasters (`dataset_ref` and `dataset_src`), or let python takes care of them, after done with them. For example,
```python
with gdal.Open(file_ref, gdal.GA_ReadOnly) as dataset_ref:
      geotransform_ref = dataset_ref.GetGeoTransform()
      nbands_ref = dataset_ref.RasterCount
      # Same for dataset_sec


```

Reply:
As a note, I think I cannot apply this comment. See: https://stackoverflow.com/questions/40513144/open-geotiff-with-gdal-produces-attributeerror-exit